### DISCUSSION TITLE
`pulumi state upgrade` scope

### DISCUSSION THREAD

**Discussion by prudnikov** [pulumi/pulumi#12958] [General]
In Pulumi v3.61.0 state scoping was changed. I'd like to migrate project by project to the new structure just not to have long stack names.

We use a custom S3 backend for the pulumi state. We have lots of projects with lots of environments each. 

[Documentation](https://www.pulumi.com/docs/intro/concepts/state/#scoping) states:
> This command will upgrade all stacks in the backend to be scoped by the project.

I expect that `pulumi state upgrade` should be executed inside one of the projects, and state scoping will change only for the current project, but this is still unclear to me. Can you please confirm that my expectation is correct? Or will it update scoping for all projects in the organization?

**Comment by abhinav** [comment]
@prudnikov Thanks for the question!
`pulumi state upgrade` will upgrade all stacks found in the backend, not just the current project.
It'll determine the project for each stack based on the project names recorded in the resource URNs—these are picked from each project separately.

**Comment by prudnikov** [comment]
@abhinav Thanks for the answer.

Just to make sure migration path is clear to me (and hopefully to anyone who will be on the same migration process) imagine the following situation:
Project 1 with name `project1`, stack name `dev-project1`, `prod-project1`
Project 2 with name `project2`, stack name `dev-project2`, `prod-project2`

We migrate to the new naming format with `pulumi state upgrade`. All the stack references will still work as before, `dev-project1`, `prod-project1`, `dev-project2`, `prod-project2` but can also be referenced as `organization/project1/dev-project1`, etc.

Now I can rename stack  `dev-project1` -> `dev`, `prod-project1` -> `prod` for project1, I MUST also change all the references from external projects/stacks to `organization/project1/dev`, and all other project will still be on the old naming method, they are not affected by the rename in `project1`.
